Sources/FunctionalKit/ReaderType.swift

```diff
@@ -69,6 +69,24 @@ extension ReaderType {
 	public func contramap <T> (_ transform: @escaping (T) -> EnvironmentType) -> Reader<T,ParameterType> {
 		return dimap(from: transform, to: { $0 })
 	}
+    
+    public static func lift<A>(_ function: @escaping (ParameterType) -> A) -> (Reader<EnvironmentType, ParameterType>) -> Reader<EnvironmentType, A> {
+        return { $0.map(function) }
+    }
+    
+    public static func lift2<A,B>(_ function: @escaping (ParameterType, B) -> A) -> (Reader<EnvironmentType, ParameterType>, Reader<EnvironmentType, B>) -> Reader<EnvironmentType,A> {
+        return { (reader1, reader2) in
+            let fn = fcurry(function)
+            return reader2.apply(reader1.map(fn))
+        }
+    }
+    
+    public static func lift3<A,B,C>(_ function: @escaping (ParameterType, B, C) -> A) -> (Reader<EnvironmentType, ParameterType>, Reader<EnvironmentType, B>, Reader<EnvironmentType, C>) -> Reader<EnvironmentType, A> {
```

Comment:
in general I prefer the following kind of style, which favors `<*>` over `apply` (to preserve shape and allow autogeneration with Sourcery) and favors protocols over concrete types for functions' inputs:

```swift
public static func lift3<A,Applicative2,Applicative3>(_ function: @escaping (ParameterType, Applicative2.ParameterType, Applicative3.ParameterType) -> A) -> (Self, Applicative2, Applicative3) -> Reader<EnvironmentType, A> where Applicative2: ReaderType, Applicative3: ReaderType, Applicative2.EnvironmentType == EnvironmentType, Applicative3.EnvironmentType == EnvironmentType {
  return { ap1, ap2, ap3 in
    Concrete.pure(fcurry(function)) <*> ap1 <*> ap2 <*> ap3
  }
}
```

what do you think?

Reply:
Yes, much better style, I will update the PR and will add tests
Thanks for reviewing